### ISSUE TITLE
conda update permission error

### ISSUE THREAD

**Issue by gruel** [conda/conda#3448]
I update the conda version on my laptop to 4.2.6 and I have now an error to update one of my environment related to writing permission. Conda is trying to save package in the main conda environement not taking into account that I am in a specific environment.

Here the log:

```
Traceback (most recent call last):
  File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 417, in download
    with open(pp, 'wb') as fo:
PermissionError: [Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/install.py", line 405, in install
    execute_actions(actions, index, verbose=not context.quiet)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/plan.py", line 643, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/instructions.py", line 132, in execute_instructions
    cmd(state, arg)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/instructions.py", line 47, in FETCH_CMD
    fetch_pkg(state['index'][arg + '.tar.bz2'])
  File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 336, in fetch_pkg
    download(url, path, session=session, md5=info['md5'], urlstxt=True)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 438, in download
    raise CondaRuntimeError("Could not open %r for writing (%s)." % (pp, e))
conda.exceptions.CondaRuntimeError: Runtime error: Could not open '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part' for writing ([Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part').

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/miniconda/lib/python3.4/site-packages/conda/exceptions.py", line 472, in conda_exception_handler
    return_value = func(*args, **kwargs)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/main.py", line 144, in _main
    exit_code = args.func(args, p)
  File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/main_update.py", line 65, in execute
    install(args, parser, 'update')
  File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/install.py", line 420, in install
    raise CondaRuntimeError('RuntimeError: %s' % e)
conda.exceptions.CondaRuntimeError: Runtime error: RuntimeError: Runtime error: Could not open '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part' for writing ([Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part').
```

**Comment by kalefranz** [comment]
It appears your user doesn't have write permissions to `/opt/miniconda/pkgs`.  Improving conda's behavior in this situation (what we typically refer to as "multi-user support") is a high priority on the roadmap.

**Comment by gruel** [comment]
I understand but the hings is that I have access to the su user and the package I am trying to update are installed as user in my home directory but conda is trying to access the miniconda installation. See after the results of the commands.

> sudo conda update --all
[sudo] Mot de passe de gruel : 

# All requested packages already installed.
# packages in environment at /opt/miniconda:
#
conda                     4.2.7                    py34_0  
conda-env                 2.6.0                         0  
openssl                   1.0.2h                        1  
pip                       8.1.2                    py34_0  
pycosat                   0.6.1                    py34_1  
pycrypto                  2.6.1                    py34_4  
python                    3.4.5                         0  
pyyaml                    3.12                     py34_0  
readline                  6.2                           2  
requests                  2.11.1                   py34_0  
ruamel_yaml               0.11.14                  py34_0  
setuptools                27.2.0                   py34_0  
sqlite                    3.13.0                        0  
system                    5.8                           2  
tk                        8.5.18                        0  
wheel                     0.29.0                   py34_0  
xz                        5.2.2                         0  
yaml                      0.1.6                         0  
zlib                      1.2.8                         3

> source activate py3
> conda update --all

Package plan for installation in environment /home/gruel/.conda/envs/py3:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    pytz-2016.6.1              |           py35_0         177 KB
    pyyaml-3.12                |           py35_0         341 KB
    pyzmq-15.4.0               |           py35_0         800 KB
    requests-2.11.1            |           py35_0         660 KB
    setuptools-27.2.0          |           py35_0         526 KB
    clyent-1.2.2               |           py35_0          15 KB
    cycler-0.10.0              |           py35_0          11 KB
    gst-plugins-base-1.8.0     |                0         3.1 MB
    h5py-2.6.0                 |      np111py35_2         2.6 MB
    pickleshare-0.7.4          |           py35_0           9 KB
    pillow-3.3.1               |           py35_0         848 KB
    python-dateutil-2.5.3      |           py35_0         237 KB
    scipy-0.18.1               |      np111py35_0        30.4 MB
    traitlets-4.3.0            |           py35_0         124 KB
    anaconda-client-1.5.1      |           py35_0         112 KB
    ipython-5.1.0              |           py35_0         991 KB
    jupyter_core-4.2.0         |           py35_0          52 KB
    pandas-0.18.1              |      np111py35_0        14.1 MB
    qt-5.6.0                   |                0        43.9 MB
    jupyter_client-4.4.0       |           py35_0          99 KB
    nbformat-4.1.0             |           py35_0         117 KB
    pyqt-5.6.0                 |           py35_0         5.4 MB
    ipykernel-4.5.0            |           py35_0         126 KB
    glueviz-0.8.2              |           py35_0         1.2 MB
    notebook-4.2.3             |           py35_0         5.0 MB
    qtconsole-4.2.1            |           py35_1         152 KB
    scikit-image-0.12.3        |      np111py35_1        26.6 MB
    widgetsnbextension-1.2.6   |           py35_0         1.1 MB
    ipywidgets-5.2.2           |           py35_0          59 KB
    nb_anacondacloud-1.2.0     |           py35_0          23 KB
    nb_conda_kernels-2.0.0     |           py35_0          30 KB
    nb_conda-2.0.0             |           py35_0          28 KB
    _nb_ext_conf-0.3.0         |           py35_0          966 B
    nbpresent-3.0.2            |           py35_0         463 KB
    ------------------------------------------------------------
                                           Total:       139.3 MB

The following NEW packages will be INSTALLED:

    _nb_ext_conf:       0.3.0-py35_0                  
    anaconda-client:    1.5.1-py35_0                  
    clyent:             1.2.2-py35_0                  
    dbus:               1.10.10-0                     
    expat:              2.1.0-0                       
    gst-plugins-base:   1.8.0-0                       
    gstreamer:          1.8.0-0                       
    icu:                54.1-0                        
    jbig:               2.1-0                         
    libgcc:             5.2.0-0                       
    libxcb:             1.12-0                        
    nb_anacondacloud:   1.2.0-py35_0                  
    nb_conda:           2.0.0-py35_0                  
    nb_conda_kernels:   2.0.0-py35_0                  
    nbpresent:          3.0.2-py35_0                  
    requests:           2.11.1-py35_0                 
    widgetsnbextension: 1.2.6-py35_0                  

The following packages will be UPDATED:

    ipykernel:          4.3.1-py35_0                   --> 4.5.0-py35_0      
    ipython:            5.0.0-py35_0                   --> 5.1.0-py35_0      
    ipywidgets:         4.1.1-py35_0                   --> 5.2.2-py35_0      
    jupyter_client:     4.3.0-py35_0                   --> 4.4.0-py35_0      
    jupyter_core:       4.1.0-py35_0                   --> 4.2.0-py35_0      
    mistune:            0.7.2-py35_0                   --> 0.7.3-py35_0      
    nbformat:           4.0.1-py35_0                   --> 4.1.0-py35_0      
    notebook:           4.2.1-py35_0                   --> 4.2.3-py35_0      
    pickleshare:        0.7.3-py35_0                   --> 0.7.4-py35_0      
    pillow:             3.2.0-py35_1       conda-forge --> 3.3.1-py35_0      
    pyqt:               4.11.4-py35_4                  --> 5.6.0-py35_0      
    pyyaml:             3.11-py35_4                    --> 3.12-py35_0       
    pyzmq:              15.3.0-py35_0                  --> 15.4.0-py35_0     
    qt:                 4.8.7-4                        --> 5.6.0-0           
    qtconsole:          4.2.1-py35_0                   --> 4.2.1-py35_1      
    scikit-image:       0.12.3-np111py35_0 conda-forge --> 0.12.3-np111py35_1
    scipy:              0.18.0-np111py35_0             --> 0.18.1-np111py35_0
    setuptools:         23.0.0-py35_0                  --> 27.2.0-py35_0     
    traitlets:          4.2.2-py35_0                   --> 4.3.0-py35_0      

The following packages will be SUPERCEDED by a higher-priority channel:

    cycler:             0.10.0-py35_0      conda-forge --> 0.10.0-py35_0      (copy)
    glueviz:            0.8.2-py35_0       conda-forge --> 0.8.2-py35_0       (copy)
    h5py:               2.6.0-np111py35_6  conda-forge --> 2.6.0-np111py35_2 
    hdf5:               1.8.17-3           conda-forge --> 1.8.17-1          
    jpeg:               9b-0               conda-forge --> 8d-1              
    libtiff:            4.0.6-6            conda-forge --> 4.0.6-2           
    pandas:             0.18.1-np111py35_1 conda-forge --> 0.18.1-np111py35_0
    pyparsing:          2.1.8-py35_0       conda-forge --> 2.1.4-py35_0      
    python-dateutil:    2.5.3-py35_0       conda-forge --> 2.5.3-py35_0       (copy)
    pytz:               2016.6.1-py35_0    conda-forge --> 2016.6.1-py35_0    (copy)

Proceed ([y]/n)? 
Fetching packages ...
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
An unexpected error has occurred.                                                                                                                                                                                      | ETA:  --:--:--   0.00  B/s
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : linux-64
          conda version : 4.2.7
       conda is private : False
      conda-env version : 4.2.7
    conda-build version : not installed
         python version : 3.4.5.final.0
       requests version : 2.7.0
       root environment : /opt/miniconda  (read only)
    default environment : /home/gruel/.conda/envs/py3
       envs directories : /home/gruel/.conda/envs
                          /opt/miniconda/envs
          package cache : /home/gruel/.conda/envs/.pkgs
                          /opt/miniconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/linux-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
            config file : None
           offline mode : False



`$ /home/gruel/.conda/envs/py3/bin/conda update --all`




    Traceback (most recent call last):
      File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 417, in download
        with open(pp, 'wb') as fo:
    PermissionError: [Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/install.py", line 405, in install
        execute_actions(actions, index, verbose=not context.quiet)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/plan.py", line 643, in execute_actions
        inst.execute_instructions(plan, index, verbose)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/instructions.py", line 134, in execute_instructions
        cmd(state, arg)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/instructions.py", line 47, in FETCH_CMD
        fetch_pkg(state['index'][arg + '.tar.bz2'])
      File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 336, in fetch_pkg
        download(url, path, session=session, md5=info['md5'], urlstxt=True)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/fetch.py", line 438, in download
        raise CondaRuntimeError("Could not open %r for writing (%s)." % (pp, e))
    conda.exceptions.CondaRuntimeError: Runtime error: Could not open '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part' for writing ([Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part').
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/opt/miniconda/lib/python3.4/site-packages/conda/exceptions.py", line 472, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/main.py", line 144, in _main
        exit_code = args.func(args, p)
      File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/main_update.py", line 65, in execute
        install(args, parser, 'update')
      File "/opt/miniconda/lib/python3.4/site-packages/conda/cli/install.py", line 420, in install
        raise CondaRuntimeError('RuntimeError: %s' % e)
    conda.exceptions.CondaRuntimeError: Runtime error: RuntimeError: Runtime error: Could not open '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part' for writing ([Errno 13] Permission denied: '/opt/miniconda/pkgs/pytz-2016.6.1-py35_0.tar.bz2.part').

**Comment by satra** [comment]
any updates here as to when this will be fixed?

**Comment by vamega** [comment]
Just ran into this issue as well? Is there a workaround that doesn't require superuser privileges?

**Comment by kalefranz** [comment]
I think this is a multi-user issue maybe.  Or something we refer to as multi-user.  I'm working on that next week.

**Comment by mworkentine** [comment]
This has occurred for me as well.  I've had anaconda installed on a multi-user system for sometime with no concerns -  this is a recent issue.  Looks like sometimes for a local (non-root) environment the tarball is being written to the central installation instead of the user's home directory (`~/.conda/envs/.pkgs`).  

Thanks for looking into this

**Comment by IanSudbery** [comment]
At the risk of saying me too .... me too!

It worked fine in the past for me in a "multi-user" environment (large HPC cluster), but has recently stopped working.

**Comment by IanSudbery** [comment]
As far as I can tell, it only happens for those packages listed as (copy) in the manifest.

**Comment by robottwo** [comment]
Happens for me too when running a 'python ./setup.py bdist_conda,,' but strangely it occurs strictly every /other/ time I run the command.

**Comment by IanSudbery** [comment]
So I 've figured out that the problem is happening if conda detects that the requested package is already present in the local package cache (~/.conda/envs/.pkgs), but not in the global cache. For some reason, under these circumstances conda tries to write to the global package cache.

**Workaround**: For me deleting ~/.conda/envs/.pkgs (or renaming it) before running `conda update` or `conda install` fixes the problem.

**Comment by swainn** [comment]
Also encountered this error during a docs build on Read The Docs. It's not as easy to delete the cache in this environment. Any fix planned for this?

**Comment by cpcloud** [comment]
This more or less prevents conda from being safely usable by more than one person.

**Comment by kalefranz** [comment]
I'm hopeful the majority of all of these issues are resolved with the latest version of conda, which is 4.3.13. 

If you update to 4.3.13 and are still having problems, can you provide the full stack trace?

**Comment by cpcloud** [comment]
@msarahan @kalefranz Would you guys accept a patch to fix this? This is very important to fix. Later versions of conda are completely useless in an organization without this fixed.

**Comment by cpcloud** [comment]
@kalefranz just saw your comment, i'll give it a shot.

**Comment by cpcloud** [comment]
@kalefranz @msarahan  I can confirm that this works now. Along with the [conda-build fix to fall back to a user writable lock file directory](https://github.com/conda/conda-build/issues/1745), we're in business!

Thanks!

**Comment by foebu** [comment]
I've got conda 4.3.13, but `conda update anaconda` still gives me
`ERROR conda.core.link:_execute_actions(330): An error occurred while uninstalling package 'defaults::numpy-1.11.1-py35_0'.
PermissionError(13, 'Permission denied')
Attempting to roll back.
PermissionError(13, 'Permission denied')`

Am I missing something here?

**Comment by kalefranz** [comment]
If the user running conda doesn't have write permissions to paths it needs to modify in the environment, then conda can't do anything.  (We can do better with error messages though.)  When conda encounters a permissions error like this during an install/uninstall transaction, the best case scenario is for conda to roll back all disk-mutating actions its made up to that point--and that's exactly what happened.  Conda behaved correctly here.

Without your `conda info` output, I can't give you specifics, but whatever environment is active during your `conda update anaconda` command has some messed up file ownership.  This most commonly happens when you use `sudo` for commands that operate on that environment.  To work your way out, you'll probably have to use `sudo chown -R` on the environment.

**Comment by foebu** [comment]
solved with `sudo chown -R user anaconda3` where user is the username, and anaconda3 is the folder where anaconda has been installed on ubuntu 16.04.

**Comment by feixian15** [comment]
yeah,it solved my problem

**Comment by duartemolha** [comment]
This is still a problem

I have done every single possible workaround and still get the permisson error. I have uninstalled and reinstalled anaconda 3 times already

Python on my system is the latest 3.6.4
Windows 10

I have done conda clean

I have made sure anaconda folder is not read only 

NOTHING WORKS

Very frustrating

**Comment by Saripalle** [comment]
w.r.t - PermissionError(13, 'Permission denied') Attempting to roll back. PermissionError(13, 'Permission denied')

Had a similar issue with the latest Anaconda installer. Reverted to older installer and it gave out the following message:

CondaIOError: Missing write permissions in: /home/ss/anaconda2
#
# You don't appear to have the necessary permissions to update packages
# into the install area '/home/user_name/anaconda2'.
# However you can clone this environment into your home directory and
# then make changes to it.
# This may be done using the command:
#
# $ conda create -n my_root --clone="/home/user_name/anaconda2"

Once I created a new environment, I could install all libraries seamlessly. It is possible that you can clone the latest environment too, but did not try that.

**Comment by BornaMD** [comment]
> Preparing transaction: done
> Verifying transaction: done
> Executing transaction: failed
> ERROR conda.core.link:_execute(481): An error occurred while uninstalling package 'defaults::conda-4.4.10-py36_0'.
> PermissionError(13, 'Access is denied')
> Attempting to roll back.
> 
> Rolling back transaction: done
> 
> PermissionError(13, 'Access is denied')

I have got the same problem and any solutions mentioned above do not work for me unfortunately.
Including the solution @Saripalle mentioned 17 days ago....
I have also tried reinstalling and emptying the .conda folder. 
Is there any way to solve this problem?

**Comment by duartemolha** [comment]
the only way I was able to get around this was to install anaconda as a single user and not for all users (windows 10)

**Comment by HyunjiEllenPak** [comment]
If you are in win10, just execute "Command Prompt (Admin)".
It solves the problem.

**Comment by duartemolha** [comment]
No it does not

**Comment by 1rdsk1** [comment]
Hi, I have this problems too.  How can I tell if my conda install is multiuser? I cant recall which option I selected.

**Comment by Saripalle** [comment]
@BornaMD just checking back on the status. Did you activate the new environment you created, and then install dependencies there? (source activate _myroot_)

**Comment by 1rdsk1** [comment]
I just unistalled anaconda and reinstalled as single user and it now updates ok.

**Comment by duartemolha** [comment]
exactly ... that is the only solution I found  no other workaround "Works"
if you install it as multiuser (on windows 10 - I have not tested on
windows 7/8)




[image: --]
Duarte Molha
[image: https://]about.me/duarte
<https://about.me/duarte?promo=email_sig>

**Comment by mahadi** [comment]
Could solve this problem on windows 7 with a simple restart. I also have the single user installation.

**Comment by Austrianguy** [comment]
Running Anaconda Prompt with admin rights solved it for me.

**Comment by fredrikaverpil** [comment]
Multiuser install of conda here on Windows 10. Running a Python subprocess with e.g. `conda update conda` fails with permission error 😢

**Comment by kalefranz** [comment]
@fredrikaverpil Because https://github.com/conda/conda/issues/3448#issuecomment-282320927, and the details of the issue are for conda 4.2.  Quite a bit has changed since then, including better (but not perfect) logic to help with issues such as these.  If you're having a problem with a recent version of conda, please file a new issue, and include the details as requested by the issue template.

**Comment by fredrikaverpil** [comment]
@kalefranz ok, will do!

**Comment by fredrikaverpil** [comment]
@kalefranz I added my info to an existing issue: https://github.com/conda/conda/issues/7100#issuecomment-383499607

**Comment by ranjankumar001** [comment]
(base) C:\Users\xyz>conda update -n base conda
Solving environment: done

## Package Plan ##

  environment location: C:\ProgramData\Anaconda3

  added / updated specs:
    - conda


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.5.4                |           py36_0         1.0 MB

The following packages will be UPDATED:

    conda: 4.4.10-py36_0 --> 4.5.4-py36_0

Proceed ([y]/n)? y


Downloading and Extracting Packages
conda 4.5.4: ########################################################## | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: failed
ERROR conda.core.link:_execute(481): An error occurred while uninstalling packag
e 'defaults::conda-4.4.10-py36_0'.
PermissionError(13, 'Access is denied')
Attempting to roll back.

Rolling back transaction: done

PermissionError(13, 'Access is denied')












please solve my problem

**Comment by xsilvertom** [comment]
Resolved 'conda update conda' on Win10 multiuser Anaconda installation by running Anaconda Prompt as Administrator.
Anaconda Prompt run normally had the same issue, regardless of a fact that Anaconda was installed as multiuser by me (and I'm local Admin as well). It seems to me that maybe UAC of Win10 participate in the issue. But I was lazy to t-shoot it deeper.

**Comment by duartemolha** [comment]
This did not work for me either

**Comment by austinlasseter** [comment]
I am experiencing this same problem.

**Comment by mesandipdatta** [comment]
For windows machine , there are workarounds to avoid Permission issues . 
Go to - 
> C:\Windows\System32
> Right click on "cmd" and choose "Run in Elevated mode"
> Opens "cmd" prompt in elevated mode.
> Type in conda commands - e.g conda install anything_you_want

Worked for me

**Comment by Rainz86** [comment]
you can also run anaconda prompt as administrator. That solved the issue for me

**Comment by amracel** [comment]
I'm in a Windows 10 environment. I'm running into this issue when I try to upgrade babel. I don't have the problem with any other installation. I elevated the privileges and it didn't solve the situation.

**Comment by Kogam22** [comment]
Getting this error after multiple times install/uninstall : 

```
kogam22@Home-L:~$ conda install anaconda-clean
Solving environment: done

## Package Plan ##

  environment location: /home/yahyaa/anaconda3

  added / updated specs: 
    - anaconda-clean


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    anaconda-clean-1.1.0       |           py36_0           5 KB

The following NEW packages will be INSTALLED:

    anaconda-clean: 1.1.0-py36_0

The following packages will be UPDATED:

    conda:          4.5.4-py36_0 --> 4.5.8-py36_0

Proceed ([y]/n)? y


Downloading and Extracting Packages
anaconda-clean-1.1.0 |    5 KB | ####################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: failed
ERROR conda.core.link:_execute(502): An error occurred while uninstalling package 'defaults::conda-4.5.4-py36_0'.
PermissionError(13, 'Permission denied')
Attempting to roll back.

Rolling back transaction: done

PermissionError(13, 'Permission denied')



```

**Comment by Kogam22** [comment]
Getting this error after multiple times install/uninstall :
```

kogam22@Home-L:~$ conda install anaconda-clean
Solving environment: done

## Package Plan ##

  environment location: /home/kogam22/anaconda3

  added / updated specs: 
    - anaconda-clean


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    anaconda-clean-1.1.0       |           py36_0           5 KB

The following NEW packages will be INSTALLED:

    anaconda-clean: 1.1.0-py36_0

The following packages will be UPDATED:

    conda:          4.5.4-py36_0 --> 4.5.8-py36_0

Proceed ([y]/n)? y


Downloading and Extracting Packages
anaconda-clean-1.1.0 |    5 KB | ####################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: failed
ERROR conda.core.link:_execute(502): An error occurred while uninstalling package 'defaults::conda-4.5.4-py36_0'.
PermissionError(13, 'Permission denied')
Attempting to roll back.

Rolling back transaction: done

PermissionError(13, 'Permission denied')




```

**Comment by MuraliElan** [comment]
It is because user does Not have permission.

If you are using windows 10 , enter anaconda in search bar, you will see anaconda prompt, right click on it and select  Run as administrator. Then proceed with command, it should work!


Cheers!

**Comment by duartemolha** [comment]
Muralidhar

If you read previous comments you will become aware this is not a fix.The
issue remains even if run with  administrator privileges

**Comment by ivan-krukov** [comment]
I ran into this issue when I was trying to install a package (`mkl`) on a fresh install of `miniconda3`. 

On this machine, there is a global installation of `anaconda` (at `/global/software/.../...`), and I have my local `miniconda` in `$HOME`. Running `conda install` incorrectly picks up the global environment, and tries to install there. This of course fails:

```
Executing transaction: failed
ERROR conda.core.link:_execute(502): An error occurred while uninstalling package 'defaults::certifi-2017.11.5-py27h71e7faf_0'.
PermissionError(13, 'Permission denied')
Attempting to roll back.

Rolling back transaction: done

PermissionError(13, 'Permission denied')
```

First, I created a new environment, then activated/deactivated it:

```
conda create --name dummy
source activate dummy
source deactivate
```

I then was able to run `conda update -n base conda`, which correctly identified my environment location:

```
Solving environment: done

## Package Plan ##

  environment location: /home/<username>/miniconda3
```

From here on, installing packages is working correctly.

**Comment by isdsava** [comment]
For those searching for this error running Windows 10:

`An error occurred while uninstalling package 'defaults::anaconda-5.2.0-py36_3'. PermissionError(13, 'Access is denied') Attempting to roll back.
`

Muralidhar's comments https://github.com/conda/conda/issues/3448#issuecomment-406851698
  helped me- running as administrator the anaconda prompt -  was successful

**Comment by mahadi** [comment]
For me also the administrator prompt solved this (windows).

**Comment by mohammedfayyad** [comment]
The same thing occurred with me on **Windows**
all the updates downloaded but at the end this message appeared
**

Preparing transaction: done
Verifying transaction: done
Executing transaction: failed
ERROR conda.core.link:_execute(502): An error occurred while uninstalling package 'defaults::anaconda-5.3.0-py37_0'.
PermissionError(13, 'Access is denied')
Attempting to roll back.

Rolling back transaction: done

PermissionError(13, 'Access is denied')

**

**Comment by lucas-alamis** [comment]
To solve this error, just run Anaconda prompt as an administrator.

**Comment by lucas-alamis** [comment]
Dear @duartemolha,

I've posted this solution because I had the same problem and worked for me.

If it worked for me, Yes my friend, that is a solution.

Best.

**Comment by duartemolha** [comment]
please dont take it the wrong way.

I am pleased that is a solution for you... but that is not a solution for
the error. There are a lot of people that work in limited systems with
limited permissions and anaconda , rightly has supported those useds by
allowing a non-admin instalation.
this  bug stops this so it needs a solution




[image: --]
Duarte Molha
[image: https://]about.me/duarte
<https://about.me/duarte?promo=email_sig>

**Comment by lucas-alamis** [comment]
Oh no, I really didn't man!

I just wrote that way because it took me several days to solve it.

I've find this shortcut at a forum, and I thought this could help.

I had to do it out of the company network, looks like I don't really had
permission to do it there.

Sorry about how I had expressed.

Em seg, 26 de nov de 2018 21:09, Duarte <notifications@github.com escreveu:

**Comment by neonb88** [comment]
I needed to check that the directory I was working with was owned by me too:

    sudo chown -R [username] ../

**Comment by mariolovric** [comment]
Tried to install jupyter with conda, have the same error now:


ERROR conda.core.link:_execute(507): An error occurred while installing package 'conda-forge::jupyter_core-4.4.0-py_0'.
PermissionError(13, 'Permission denied')

**Comment by aditya-pillai009** [comment]
I had the error PermissionError(13, 'Permission denied') while installing tensorflow.
All I did was run Anaconda prompt as administrator and execute the same commands again and it worked.

**Comment by AllisonLiu16** [comment]
> This is still a problem
> 
> I have done every single possible workaround and still get the permisson error. I have uninstalled and reinstalled anaconda 3 times already
> 
> Python on my system is the latest 3.6.4
> Windows 10
> 
> I have done conda clean
> 
> I have made sure anaconda folder is not read only
> 
> NOTHING WORKS
> 
> Very frustrating

The main point for this problem is that anacond in your computer has no permission as a root user's. If you change anaconda's permission to root and group belongs to admin. I think it will be worked. Anyway it works for me.

**Comment by Geomaan** [comment]
> > This is still a problem
> > I have done every single possible workaround and still get the permisson error. I have uninstalled and reinstalled anaconda 3 times already
> > Python on my system is the latest 3.6.4
> > Windows 10
> > I have done conda clean
> > I have made sure anaconda folder is not read only
> > NOTHING WORKS
> > Very frustrating
> 
> The main point for this problem is that anacond in your computer has no permission as a root user's. If you change anaconda's permission to root and group belongs to admin. I think it will be worked. Anyway it works for me.
-----------------------------------------------------------------
This worked for me:
Resolved multiuser problem on Win10  by running cmd (c: windows\system32\cmd) Prompt as Administrator.

**Comment by jimmyknowles** [comment]
> So I 've figured out that the problem is happening if conda detects that the requested package is already present in the local package cache (~/.conda/envs/.pkgs), but not in the global cache. For some reason, under these circumstances conda tries to write to the global package cache.
> 
> **Workaround**: For me deleting ~/.conda/envs/.pkgs (or renaming it) before running `conda update` or `conda install` fixes the problem.

This worked for me as well, spent several hours trying to figure out the permission denied error, but as soon as I cleared out the cache of my cloned env everything started working normally again. Thanks for your help.

Not sure why conda wants to write to the cache of a non-active env...sounds like a bug to me. Or should give more meaningful feedback.

**Comment by Didou09** [comment]
Hi,

I do research, and like many of my colleagues I work, as a user with no admin right, on a shared linux cluster.
There is a common base environment that everyone can use.
There is also a few shared environements that eveyrone can use too

And I can, create my own environements, locally.
In the example below, the first 3 are shared environments managed by the admin.
The next two (empty and tofu141) were created by me, for my own needs.

```
(empty) [ DV226270 spica ~/ToFu_All/tofu_git/tofu (master)] conda info --envs
# conda environments:
#
base                     /Applications/Anaconda/python36
test                     /Applications/Anaconda/python36/envs/test
tofu-1.4.1               /Applications/Anaconda/python36/envs/tofu-1.4.1
empty                 *  /Home/DV226270/.conda/envs/empty
tofu141                  /Home/DV226270/.conda/envs/tofu141
```

When I try to install packages on any of my local environments (e.g.: 'empty'), I get the following and the installation does not even start:

```
(empty) [ DV226270 spica ~/ToFu_All/tofu_git/tofu (master)] conda install -n empty conda-build numpy scipy matplotlib cython>=0.26


==> WARNING: A newer version of conda exists. <==
  current version: 4.7.11
  latest version: 4.8.0rc0

Please update conda by running

    $ conda update -n base -c defaults conda
```
 

I have two problems here:
* Fisrt, I don't have admin access to the base environment, so I can't update conda, which in practice prevents me from installing anything because conda does not seem to be able to get passed that warning.
* Second, I guess having environments in 2 separate places means I have 2 cache, is there a way to specify which cache I want to erase when using command `conda clean --all` ? Because in practice I don't have access to the shared cache and it takes a loooong time for conda to get through all packages and tarballs in there and throw '_Permission denied_' errors at each one of them...

For general information:
```
(empty) [ DV226270 spica ~/ToFu_All/tofu_git/tofu (master)] conda info --envs
# conda environments:
#
base                     /Applications/Anaconda/python36
test                     /Applications/Anaconda/python36/envs/test
tofu-1.4.1               /Applications/Anaconda/python36/envs/tofu-1.4.1
empty                 *  /Home/DV226270/.conda/envs/empty
tofu141                  /Home/DV226270/.conda/envs/tofu141

(empty) [ DV226270 spica ~/ToFu_All/tofu_git/tofu (master)] conda info

     active environment : empty
    active env location : /Home/DV226270/.conda/envs/empty
            shell level : 2
       user config file : /Home/DV226270/.condarc
 populated config files : /Home/DV226270/.condarc
          conda version : 4.7.11
    conda-build version : 3.4.2
         python version : 3.6.9.final.0
       virtual packages : 
       base environment : /Applications/Anaconda/python36  (read only)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/Didou09/linux-64
                          https://conda.anaconda.org/Didou09/noarch
                          https://conda.anaconda.org/tofuproject/linux-64
                          https://conda.anaconda.org/tofuproject/noarch
          package cache : /Applications/Anaconda/python36/pkgs
                          /Home/DV226270/.conda/pkgs
       envs directories : /Home/DV226270/.conda/envs
                          /Applications/Anaconda/python36/envs
               platform : linux-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.6.9 Linux/2.6.32-754.10.1.el6.x86_64 centos/6.10 glibc/2.12
                UID:GID : 14013:100
             netrc file : None
           offline mode : False
```

**Comment by wulmer** [comment]
Hi @Didou09, please open a new issue next time, especially if the current one is already closed.

Regarding your problems: The warning message about a conda update is no error, you can simply ignore it. The problem that you don't see anything happening then is that you pipe your command output into a file `=0.26`, as your command line is something like `conda install -n empty conda-build numpy scipy matplotlib cython>=0.26`.  The larger than sign makes all command line output disappear in the specified file. Try putting the last argument in single quotes, e.g. `conda install -n empty conda-build numpy scipy matplotlib 'cython>=0.26'`.

About the package cache issue: If you look closely at the [configuration options](https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-environment-directories-envs-dirs) of your `.condarc` file, you'll find two which might help you in defining your main package cache directory: `envs_dirs` or `pkgs_dirs`. I don't have anything available right now for testing, but I had a similar issue some years ago and I was able to redefine my package cache to my private directory.

Good luck!
